pkg/workspace/library_module.go

```diff
@@ -58,13 +80,76 @@ func (l LibraryModule) Get(thread *starlark.Thread, f *starlark.Builtin,
 
 	libraryCtx := LibraryExecutionContext{Current: foundLib, Root: foundLib}
 
-	return (&libraryValue{libPath, libraryCtx, nil, l.libraryExecutionFactory}).AsStarlarkValue(), nil
+	beforeLibModValuess, afterLibModValuess, childLibValueDocs, err := l.getValuesForLibraryAndChildren(
+		l.libraryValues,
+		libraryCtx.Current.name,
+		libTag,
+	)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	return (&libraryValue{
+		libPath,
+		libTag,
+		libraryCtx,
+		beforeLibModValuess,
+		afterLibModValuess,
+		childLibValueDocs,
+		l.libraryExecutionFactory,
+	}).AsStarlarkValue(), nil
+}
+
+const (
+	ChildLib int = iota
+	CurrentLib
+	OtherLib
+)
+
+func (l LibraryModule) getValuesForLibraryAndChildren(valueDocs []*DataValues,
+	currentLibName, libTag string) ([]*DataValues, []*DataValues, []*DataValues, error) {
+
+	var currentBeforeModValues []*DataValues
+	var currentAfterModValues []*DataValues
+	var childLibValues []*DataValues
+
+	for _, doc := range valueDocs {
+		copiedDataValues := &DataValues{Doc: doc.Doc, LibPath: doc.LibPath, AfterLibMod: doc.AfterLibMod}
```

Comment:
doc.Doc could be deep copied (may be just in case?), but doc.LibPath seems to be somewhat dangerous to pass around like that since it is just pointing to original slice. probably best to make a DataValues.DeepCopy() method and do some work there.